test/server/cards/03_TWI/events/PetitionTheSenate.spec.ts

```diff
@@ -0,0 +1,27 @@
+describe('Petition The Senate', function () {
+    integration(function (contextRef) {
+        describe('Petition The Senate\'s ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['petition-the-senate'],
+                        groundArena: ['general-tagge#concerned-commander',
+                            'the-client#dictated-by-discretion',
+                            'colonel-yularen#isb-director']
+                    },
+                    player2: {
+                        groundArena: ['atst'],
+                        spaceArena: ['green-squadron-awing']
+                    }
+                });
+            });
+
+            it('If control 3 Official units, draw 3 cards', function () {
+                const { context } = contextRef;
+                context.player1.clickCard(context.petitionTheSenate);
+                expect(context.player1.hand.length).toBe(3);
+            });
```

Comment:
Need a second test with only two Official units in play to confirm that nothing happens. Ideally there would also be an opponent Official unit in play to confirm that isn't being counted as part of the 3

Reply:
Added a new test and when an official is defeated and click is done on card I get - Expected player prompt state to change after clicking petition-the-senate but it did not.

Is this correct, or should card go to Discard with no outcome?

Reply:
Pinged in discord as well, but having trouble understanding the situation you describe - can you switch this to draft and push the broken test so I can see.

Also, if you have open comments that I need to answer please leave them unresolved so I can find them easier. Thanks!